src/position_manager.rs

```diff
@@ -0,0 +1,517 @@
+use core::fmt::Debug;
+
+use crate::{
+    abi::{self, PoolKey},
+    prelude::Position,
+    utils::{
+        abi as Interface, v4_postition_planner::V4PositionPlanner,
+        V4Planner,
+    },
+};
+
+use super::entities::{position, Pool};
+use alloc::{boxed::Box, string::String, vec::Vec};
+use alloy_primitives::{aliases::U48, Address, Bytes, Signature, I256, U128, U160, U256};
+use alloy_sol_types::{
+    abi::{encode, encode_params, token::DynSeqToken},
+    SolValue,
+};
+use uniswap_sdk_core::{
+    entities::native_currency,
+    prelude::{
+        BaseCurrencyCore, BigDecimal, BigInt, Currency, FractionBase, NativeCurrency, Percent, Zero,
+    },
+};
+use uniswap_v3_sdk::{multicall, prelude::Multicall, utils::MethodParameters};
+
+pub trait CommonOptions {
+    fn slippage_tolerance(&self) -> &Percent;
+    fn hook_data(&self) -> Option<Bytes>;
+    fn deadline(&self) -> U256;
+}
+
+pub trait ModifyPositionSpecificOptions {
+    fn token_id(&self) -> U256;
+}
+
+pub trait MintSpecificOptions {
+    fn recipient(&self) -> Address;
+    fn create_pool(&self) -> Option<bool>;
+    fn sqrt_price_x96(&self) -> Option<U256>;
+    fn migrate(&self) -> Option<bool>;
+}
+
+pub trait CommonAddLiquidityOptions {
+    fn use_native(&self) -> Option<&Currency>;
+    fn batch_permit(&self) -> Option<&BatchPermitOptions>;
+}
+
+pub trait CollectSpecificOptions {
+    fn token_id(&self) -> U256;
+    fn recipient(&self) -> Address;
+}
+
+pub trait TransferOption {
+    fn sender(&self) -> Address;
+    fn recipient(&self) -> Address;
+    fn token_id(&self) -> U256;
+}
+
+pub trait RemoveLiquiditySpecificOptions {
+    fn liquidity_percentage(&self) -> &Percent;
+    fn burn_token(&self) -> Option<bool>;
+    fn permit(&self) -> Option<&NFTPermitOptions>;
+}
```

Comment:
Remove all traits and impls, aka TypeScript garbage. Follow the approach in `uniswap-v3-sdk-rs`.

Reply:
hate typescript so much